src/yetibot/commands/weather.clj

```diff
@@ -44,13 +44,19 @@
   [path pc cc]
   (get-json (endpoint path) {:query-params {:postal_code pc
                                             :country cc}}))
+
 (defn- get-by-loc
   "Attempt to parse out postal code and call the corresponding get-by-name or
    get-by-pc function"
   [path loc]
-  (if-let [[pc cc] (apply chk-postal-code (str/split (str loc) #"\s*,\s*"))]
-    (get-by-pc path pc cc)
-    (get-by-name path loc)))
+  (let [[_ pc cc] (re-matches #"(.+?)(?:,\s*([^,]+))?" (str loc))]
```

Comment:
I think I want to see the branches to do either:
1. postal code + country code or
1. name based lookups

once, at the top. The regex becomes a little simper too if we're only attempting to detect the postal code case.

What do you think of something like this?

```clojure
  (let [path "current"
        loc "59101"
        [_ pc cc] (re-matches #"(\d{4,})(?:,\s*([^,]+))?" (str loc))]
    (if pc
      (apply get-by-pc path (apply chk-postal-code (remove nil? [pc cc])))
      (get-by-name path loc)))
```

Not sure if my regex is correct for minimum viable zip code, but I tried and verified it for these values of `loc`:

- 59101
- 59101, US
- Seattle
- Seattle, WA, US
- Toronto, Ontario, CA